client/layout/masterbar/checkout.tsx

```diff
@@ -92,8 +124,26 @@ const CheckoutMasterbar: FunctionComponent< Props > = ( {
 				<span className="masterbar__secure-checkout-text">{ translate( 'Secure checkout' ) }</span>
 			</div>
 			<Item className="masterbar__item-title">{ title }</Item>
+			<CheckoutModal
+				title={ modalTitleText }
+				copy={ modalBodyText }
+				closeModal={ () => setIsModalVisible( false ) }
+				isVisible={ isModalVisible }
+				buttonCTA={ modalPrimaryText }
+				primaryAction={ leaveCheckout }
+				secondaryButtonCTA={ modalSecondaryText }
+				secondaryAction={ clearCartAndLeave }
+			/>
 		</Masterbar>
 	);
 };
 
-export default CheckoutMasterbar;
+export default function CheckoutMasterbarWrapper( props: Props ): JSX.Element {
+	return (
+		<CalypsoShoppingCartProvider>
+			<ThemeProvider theme={ checkoutTheme }>
+				<CheckoutMasterbar { ...props } />
+			</ThemeProvider>
```

Comment:
The `ThemeProvider` is needed to color the modal text with `theme.colors.textColor` and nothing else?

Reply:
Also `theme.colors.modalBackground`, `theme.colors.surface`, `theme.colors.borderColorLight`, `theme.weights.normal`.

I'd prefer not to need `CheckoutModal` here and to use a native modal from elsewhere in `client`, but I was unable to find one that already has styling applied. I would have to build the styles from scratch and this seemed a lot less work. However, I'm very open to ideas for what other options we could use.

I also considered changing the styles so that there were defaults applied if a `ThemeProvider` was not present, but that felt like too big a change for this PR, which is why I ended up just adding the `ThemeProvider`. Since this is only shown on the checkout page, which already loads a `ThemeProvider`, I figure that it won't increase the bundle size anywhere else.